### PR TITLE
Merging latest changes before end of TWIP

### LINKED ISSUE #5: Implement Domain Randomization (DR)
To get a more encompassing understanding of the real world, we need to implement DR for training the TWIPs.

Not sure if parameters should be randomized on reset only or at random intervals.

### LINKED ISSUE #11: Add noise in the IMU output values
We should definitely add noise into the IMU implementation: it'll improve reliability and lower the sim to real gap!

Is there a way to add noise in the imu values? Not sure if this one but it seems like an ideal IMU. Real ones have noise.

_Originally posted by @Shamiivan in https://github.com/Sim2Capstone/twip-isaac-sim/pull/2#discussion_r1688206102_

### LINKED ISSUE #21: Implement random reset orientations
TWIP currently starts from the same fixed orientation when training: we should make it randomize to account for more starting conditions

### PR DESCRIPTION
**Pull request by AugustDG**
We had a bunch of features that merge into this PR, most are chores, but the following correspond to an issue:
- Closes #21 
- Closes #11 
- Closes #5 
- #20

### CODE REVIEW

File: core/twip/assets/.thumbs/256x256/twip.usd.png
Comment (Shamiivan) [inline core/twip/assets/.thumbs/256x256/twip.usd.png:1]: Depending on whether we publish this or not, we should keep the usd cleaner(ie keep the original one)

File: core/twip/balancing_twip_task.py [hunk 1]
```diff
@@ -74,7 +80,7 @@ def __init__(
         )
 
         reward_space = gymnasium.spaces.Box(
-            low=np.array([-1.0]),
+            low=np.array([-2.0]),
```
Comment (Shamiivan) [inline core/twip/balancing_twip_task.py:83]: why -2?

**Review by Shamiivan — APPROVED** [review]
LGTM!

File: core/twip/balancing_twip_task.py (hunk 1 above)
Comment (AugustDG) [reply core/twip/balancing_twip_task.py:83]: A bit below is written the penalty when you reset

File: core/twip/assets/.thumbs/256x256/twip.usd.png
Comment (AugustDG) [reply core/twip/assets/.thumbs/256x256/twip.usd.png:1]: The cleaner one? What do you mean?

Comment (Shamiivan) [reply core/twip/assets/.thumbs/256x256/twip.usd.png:1]: Cleaner meaning original one

Comment (AugustDG) [reply core/twip/assets/.thumbs/256x256/twip.usd.png:1]: They didn't have one iirc, only the URDF

**Pull request merged**